notation.go

```diff
@@ -166,7 +168,12 @@ func Sign(ctx context.Context, signer Signer, repo registry.Repository, signOpts
 	logger.Debugf("Pushing signature of artifact descriptor: %+v, signature media type: %v", targetDesc, signOpts.SignatureMediaType)
 	_, _, err = repo.PushSignature(ctx, signOpts.SignatureMediaType, sig, targetDesc, annotations)
 	if err != nil {
-		logger.Error("Failed to push the signature")
+		var referrerError *remote.ReferrersError
+		if errors.As(err, &referrerError) && referrerError.IsReferrersIndexDelete() {
+			logger.Warn("The signature has been attached to the artifact, but it failed to delete the dangling referrers index.")
+		} else {
+			logger.Error("Failed to push the signature")
```

Comment:
nit:
We need to align our log format. I'm seeing some log ends with a `.`, others don't. Which format is desired here?
And we'd like all logs to begin with a capital letter.

Reply:
Removed `.`